### PR TITLE
Removed some vestigial code related to Range bounds checks

### PR DESCRIPTION
**Pull request by philip-peterson**
When an LFS pointer file has a size that is too large (compared to the actual blob size), the Git LFS client can make Range requests that are out of bounds. Currently Gitea tries to obey these requests but gets an EOF during the copy and still responds with a 200, so this PR adds a bounds check to error early if that would be about to happen.

Also removed some vestigial code related to the bounds checks.

Update: see comment below

### CODE REVIEW

File: services/lfs/server.go [hunk 1]
```diff
@@ -131,6 +131,21 @@ func DownloadHandler(ctx *context.Context) {
...
+	stat, err := content.Stat()
+	if err != nil {
+		log.Error("Unable to stat LFS OID[%s]. Error: %v", meta.Oid, err)
+
+		writeStatus(ctx, http.StatusInternalServerError)
+		return
+	}
+
+	if toByte+1 > stat.Size() {
+		log.Error("Whilst trying to read LFS OID[%s]: Unable to read to %d Error: %v", meta.Oid, toByte, err)
```
Comment (Gusted) [inline services/lfs/server.go]: Maybe explicitly mentioned that out-of-range was requested.

**Review by Gusted — COMMENTED** [review]
Is there a test case where this issue can be reproduced(and ideally can be included into Gitea integration tests.

**Comment by KN4CK3R** [comment]
I don't understand the problem. The initial `toByte = meta.Size - 1` (`meta.Size` is the content size, there should be no need to stat it). If the client provides a range, `toByte` gets capped to this size. How can it be out of bounds?

**Comment by wxiaoguang** [comment]
It comes from user's HTTP header.

https://github.com/go-gitea/gitea/pull/20312/files#diff-95b595c61d15024441878562cf12099f39f68773a995de51ff021f7258b5a695R92-L114


While I think the check should be moved above, otherwise the response header (https://github.com/go-gitea/gitea/pull/20312/files#diff-95b595c61d15024441878562cf12099f39f68773a995de51ff021f7258b5a695R111) is still incorrect.

**Comment by KN4CK3R** [comment]
The meta size is checked against the written size, must be correct.

https://github.com/go-gitea/gitea/blob/cb6c5f8193946cf65ba9b325a1989d72faaf2b86/modules/lfs/content_store.go#L142-L144

**Comment by philip-peterson** [comment]
Yes, I will have to check to see if this is a problem in latest Gitea. The hashingReader code looks like it might address it, I was working based off an older version.

**Comment by philip-peterson** [comment]
I was able to verify that this issue has been fixed in latest Gitea. Although, there is still some dead code, so I changed this PR to be just about removing the dead code.

Thanks for the help, all.

**Review by wxiaoguang — APPROVED** [review]

**Review by Gusted — APPROVED** [review]

**Pull request merged**